internal/expr/parser.go

```diff
@@ -37,20 +37,9 @@ func (p *Parser) Parse(input string) (pe *ParsedExpr, err error) {
 		}
 	}()
 
-	p.init(input)
-
-	for {
-		// Advance the parser to the start of the next expression.
-		if err := p.advance(); err != nil {
-			return nil, err
-		}
-
-		p.currentExprStart = p.pos
-
-		if p.pos == len(p.input) {
-			break
-		}
+	p.init(input, 0)
```

Comment:
The only reason it is here is because I dont want anyone to have to debug it when the need arises later. It is not trivial to see that you need to set all three variables and keep the input for error reporting purposes (in fact we made the same mistake). Think about adding scaffolding so that us or other people do not have to re-do the investigation later.